non-blocking set for wrap function

corradodellorusso opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] Followed the [Contributing](https://github.com/jaredwray/cache-manager/blob/main/CONTRIBUTING.md) guidelines.
- [x] Tests for the changes have been added (for bug fixes/features) with 100% code coverage.
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Introducing non-blocking set for wrap operations. 
In V4 when the wrap operation was called, results were returned before backfilling the cache(s).
This changed in V5 and cache(s) are backfilled before returning results, leading to slower throughput. 

This PR adds the option to restore the behaviour of V4.

jaredwray commented:
@corradodellorusso  would we want to extend this to the entire library or just wrap?

corradodellorusso commented:
> @corradodellorusso would we want to extend this to the entire library or just wrap?

@jaredwray as far as I understand `wrap` is the only operation that involves multiple different operations in its implementation, for the others operation is sufficient to omit `await` to obtain the same result.

jaredwray commented:
@corradodellorusso - we have added nonBlocking as a core but have not implemented it in wrap yet. Do you still need that?

corradodellorusso commented:
Hello Jared, I’m not sure I got the question.In the version we currently use (5.7.6) we implemented the nonBlockingSet as a parameter for wrap on the single cache.Does this answer your question?Inviato da iPhoneIl giorno 25 set 2024, alle ore 18:05, Jared Wray ***@***.***> ha scritto:﻿
@corradodellorusso - we have added nonBlocking as a core but have not implemented it in wrap yet. Do you still need that?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

jaredwray commented:
Thanks. We added non blocking functionality across the rest of the service also with v6
